controllers/cloudstackmachinestatechecker_controller.go

```diff
@@ -81,21 +81,23 @@ func (r *CloudStackMachineStateCheckerReconciliationRunner) Reconcile() (ctrl.Re
 
 	csRunning := r.CSMachine.Status.InstanceState == "Running"
 	csTimeInState := r.CSMachine.Status.TimeSinceLastStateChange()
-	capiRunning := r.CAPIMachine.Status.Phase == "Running"
+	capiPhase := capiv1.MachinePhase(r.CAPIMachine.Status.Phase)
+	capiRunning := capiPhase == capiv1.MachinePhaseRunning
+	deleteInvoked := capiPhase == capiv1.MachinePhaseDeleting || capiPhase == capiv1.MachinePhaseDeleted
 
 	// capiTimeout indicates that a new VM is running, but it isn't reachable due to a network issue or a misconfiguration.
 	// When this happens, the machine should be deleted or else the cluster won't ever recover.
 	capiTimeout := csRunning && !capiRunning && csTimeInState > 5*time.Minute
 
 	if csRunning && capiRunning {
 		r.ReconciliationSubject.Status.Ready = true
-	} else if !csRunning || capiTimeout {
+	} else if (!csRunning || capiTimeout) && !deleteInvoked {
 		r.Log.Info("CloudStack instance in bad state",
 			"name", r.CSMachine.Name,
 			"instance-id", r.CSMachine.Spec.InstanceID,
 			"cs-state", r.CSMachine.Status.InstanceState,
 			"cs-time-in-state", csTimeInState.String(),
-			"capi-phase", r.CAPIMachine.Status.Phase)
+			"capi-phase", capiPhase)
 
 		if err := r.K8sClient.Delete(r.RequestCtx, r.CAPIMachine); err != nil {
```

Comment:
are we still sure that deleting the CAPI machine is the right thing to do? It feels a bit risky and like a shortcut. Have we considered deleting the CloudStackMachine instead, or is there something preventing us from doing that?

Reply:
FWIW, the CAPI-provided Machine Health Check mechanisms deletes CAPI machines.  Its subordinate InfrastructureMachine (i.e., CloudStackMachine) gets deleted with it.

Reply:
[PR-58](https://github.com/kubernetes-sigs/cluster-api-provider-cloudstack/pull/58) actually addressed a problem caused when we delete CloudStackMachine instead of Machine.

Reply:
/approved